sui_programmability/examples/fungible_tokens/sources/BASKET.move

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// A synthetic fungible token backed by a basket of other tokens.
+/// Here, we use a basket that is 1:1 SUI and MANAGED,
+/// but this approach would work for a basket of any composition
+module FungibleTokens::BASKET {
+    use FungibleTokens::MANAGED::MANAGED;
+    use Sui::Coin::{Self, Coin, TreasuryCap};
+    use Sui::ID::VersionedID;
+    use Sui::SUI::SUI;
+    use Sui::Transfer;
+    use Sui::TxContext::{Self, TxContext};
+
+    /// Name of the coin. By convention, this type has the same name as its parent module
+    /// and has no fields. The full type of the coin defined by this module will be `COIN<BASKET>`.
+    struct BASKET has drop { }
+
+    /// Singleton shared object holding the reserve assets and the capability.
+    struct Reserve has key {
+        id: VersionedID,
+        /// capability allowing the reserve to mint and burn BASKET
+        treasury_cap: TreasuryCap<BASKET>,
+        /// SUI coins held in the reserve
+        sui: Coin<SUI>,
+        /// MANAGED coins held in the reserve
+        managed: Coin<MANAGED>,
+    }
+
+    /// Needed to deposit a 1:1 ratio of SUI and MANAGED for minting, but deposited a different ratio
+    const EBAD_DEPOSIT_RATIO: u64 = 0;
+
+    fun init(ctx: &mut TxContext) {
+        // Get a treasury cap for the coin put it in the reserve
+        let treasury_cap = Coin::create_currency<BASKET>(BASKET{}, ctx);
+        Transfer::share_object(Reserve {
+            id: TxContext::new_id(ctx),
+            treasury_cap,
+            sui: Coin::zero<SUI>(ctx),
+            managed: Coin::zero<MANAGED>(ctx),
+        })
+    }
+
+    /// === Writes ===
+
+    /// Mint BASKET coins by accepting an equal number of SUI and MANAGED coins
```

Comment:
I don't fully understand how a Basket works. Why providing an equal number of SUI and MANAGED will generate an equal number of BASKET? Does the pricing of these coins not play a role?

Reply:
I believe this system is similar to Uniswap where you get a liquidity token for providing liquidity to one or both of the pools. It can be improved to support flexible rates for each of the currency. But that's my guess. 😆

Reply:
The idea is that the value of BASKET fluctuates with the value of the assets in the reserve.

For a real-life example, [SDR](https://www.imf.org/en/About/Factsheets/Sheets/2016/08/01/14/51/Special-Drawing-Right-SDR) is a synthetic currency composed of a basket of USD, EUR, RMB, JPY, and GBP.

Similarly, you might want a synthetic currency backed by a basket of NFT's that changes over time (e.g., https://www.coindesk.com/markets/2022/02/10/flamingodaos-nft-portfolio-is-now-worth-1b/)

I will add a comment about this.